pennylane/devices/preprocess.py

```diff
@@ -45,21 +46,37 @@ def null_postprocessing(results):
 def _operator_decomposition_gen(
     op: qml.operation.Operator,
     acceptance_function: Callable[[qml.operation.Operator], bool],
+    decomposer: Optional[
+        Callable[[qml.operation.Operator], Sequence[qml.operation.Operator]]
+    ] = None,
+    max_expansion: Optional[int] = None,
+    current_depth=0,
```

Comment:
Another totally optional comment.

I wonder if we use the difference between the two and only have one argument.

```
# initialize to np.inf if infinite recursion
depth = np.inf
# check if reached zero yet
depth < 1
# subtract one for next round
depth -= 1
```